### ISSUE TITLE
Design: How could we display downtime alerts in widgets that offer DS Logon as an authetication option?

### ISSUE THREAD

**Issue by wesrowe** [department-of-veterans-affairs/va.gov-cms#13736]
### User story

**AS A** Veteran
**I WANT** to be alerted in the context of a widget that offers DS Logon when that service will be temporarily unavailable
**SO THAT** I can understand why login options are changed.

"In the context of a widget" could mean right above it, or even inside it. Here's [an example](https://www.va.gov/health-care/refill-track-prescriptions/):
<img width="486" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/4054752/b2436eca-8fa6-49bd-8a19-7a358d69c271">

### Notes / background

Design notes:
- Pay careful attention to how the existing DTAs work. Documented in [ticket comment](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/12339#issuecomment-1536534924). 
- A modal is involved, which could be a problem if logging in by other methods (besides DSL) will still be possible.
- Also, sign-in itself happens in a modal. If we recommend a DTA in the sign-in modal (e.g., "You can log in but not with DSLogon"), it can't be a modal.
 

Engineering questions that might drive this:
- When DSL is down, are Veterans still able to use alternatives? What does that UX look like?

## Acceptance criteria

- [ ]

**Comment by randimays** [comment]
I looked through [this MURAL](https://app.mural.co/t/departmentofveteransaffairs9999/m/vagov6717/1658961045082/cedcb1c5c42489e47be37733c73890ef890070ae?sender=ue51e6049230e03c1248b5078) and [this Github doc](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/public-websites) to gather all Public Websites products. 

Many of the products are built exclusively by Drupal and `content-build`, which means they are static pages without external API calls (most only calling for feature toggles). Even if they did have external API calls, they would not able to use `vets-website`'s `<DowntimeNotification>` functionality without a corresponding React component, so we'd have to build another solution.

There is one use case, the VA.gov search, that is capable of using these notifications, but they have already been set up.

I confirmed with Platform [here](https://dsva.slack.com/archives/CBU0KDSB1/p1690394943656469) that any calls to `va.gov` or `api.va.gov` calls are not considered "external" and do not need `<DowntimeNotification>`s. If those services were down, it's likely we'd know almost immediately through another channel.

Here is the result: [Spreadsheet of PW products](https://docs.google.com/spreadsheets/d/1Q3-aJOCyIk_CjVT0rgc3Me08aXVM48bpg6uIxRbYDdI/edit?usp=sharing)

**Comment by jilladams** [comment]
Amazing. Thanks for the write up, Randi. Based on your notes above and the contents of the sheet, this makes sense to me, but I'll defer to technical review / Wes with any product-side questions before we call this Done.

**Comment by wesrowe** [comment]
I left a couple comments. Two products that I think of as depending on external services got a "No" and I think we might want to discuss definitions and the feedback Randi got from Platform. Cerner widgets and Yellow Ribbon.

**Comment by wesrowe** [comment]
It seems that we should meet as a team to determine what next steps to take, if any. This will carry over to S90.

**Comment by randimays** [comment]
@wesrowe I confirmed that all Cerner widgets except for the appointments app use MHV, and we can add Downtime Notifications for those. I am [trying to confirm with the appointment team](https://dsva.slack.com/archives/CMNQT72LX/p1690993443708019) about the appointments app. It uses `https://patientportal.myhealth.va.gov/` which I understand is the "Cerner portal," but not sure yet which external service it maps to. I had also asked Platform; they weren't quite sure.

The Yellow Ribbon tool does ultimately connect to an external service that seems to be referred to as "GIDS," but it has a vets-api gatekeeper. The only calls the Yellow Ribbon app makes are to `va.gov` services, that then reach out to GIDS. We won't be able to add a Downtime Notification here because of that middleware.

CC @jilladams

**Comment by wesrowe** [comment]
Re: Cerner widgets – all 5 use the Cerner portal for Cerner patient services – so in addition to MHV (and any scheduling component if outside MHV), the possibility of "Cerner being down" would be noteworthy to Veterans. (I don't know what the URLs are for MHV and Cerner.)

But I think the goal was to set a meeting to discuss these findings. I'll ask DaveC if he wants to join that one too, in addition to the header/footer discovery you did.

**Comment by randimays** [comment]
Just adding an update from my end here: I believe we're just waiting on a scheduled meeting to go over these results (next week?)

**Comment by wesrowe** [comment]
meeting is scheduled for Wednesday, 8/9

**Comment by wesrowe** [comment]
Note that I updated Randi's comment above to link to the [Excel version of her work on Sharepoint](https://dvagov-my.sharepoint.com/:x:/g/personal/wesley_rowe_va_gov/EUDfw0hcpPFNv857ztSVETAB3bZkmTKAQkm1SjKjpr2EmQ?e=hnOXqf).

**Comment by jilladams** [comment]
@wesrowe @randimays Re: the questions around previously trying to get pagerduty access: 
1. 3/7/23: https://dsva.slack.com/archives/CBU0KDSB1/p1678221514593359
2. https://dsva.slack.com/archives/CBU0KDSB1/p1678226056905779
3. related ticket: https://github.com/department-of-veterans-affairs/va.gov-team/issues/54755 -- account created

**Comment by randimays** [comment]
Notes from Downtime Alerts discussion:

Observations / general questions:
- Need for downtime notifications is specific to Cerner (authenticated) widgets. All other PW products are already covered or don't need them in vets-website.
- Global`user` object - dependency on Cerner service?
- If login modal has downtime notifications, would any outages here be owned by PW?
- Distinction between devs alerted vs. veterans are notified on FE
- If My HealtheVet (MHV) goes down, do we know about it?
- In the future when all EMR services live on va.gov, the process may be streamlined
- Currently, veterans have to log in to va.gov and then the external portals. If those systems are down or in scheduled maintenance, we need to alert the veteran before they try to go there.

Action items / Outstanding questions
- @wesrowe to check with Dave re:
  - If either portal goes down, should we display a downtime alert?
  - Should Public Websites own downtime notifications for the Cerner widgets specifically, or is this a larger Platform issue?
  - Should each team manage the login process (for va.gov and the external portals) for their own widget?
- How do we get these Cerner widgets into PagerDuty so we can get notifications in real time?

CC @jilladams @chriskim2311

**Comment by jilladams** [comment]
From scrum: 
* Gonna get tickets for 4 widgets that use MHV
* Appointments tool just uses Cerner, TBD on that one
* Will get in PagerDuty and look around, because we can.

**Comment by randimays** [comment]
I asked the folks in [#vagov-cerner-launch-coordination](https://dsva.slack.com/archives/C0190MTGNUE/p1691682926583289) if they have any insights about Cerner outages / scheduled maintenance, and/or if there are already downtime notifications in place or a plan to add them.

**Comment by randimays** [comment]
Adding an end-of-sprint update: no replies on the above thread yet; Jill suspects Lauren Alexanderson will know the answers to our questions, but she is on PTO until tomorrow.

**Comment by jilladams** [comment]
Noting: this work was specific to products with external service dependencies that are _not_ routed through vets-api.
